src/flamegpu/simulation/detail/AbstractSimRunner.cu

```diff
@@ -0,0 +1,114 @@
+#include "flamegpu/simulation/detail/AbstractSimRunner.h"
+
+#ifdef _MSC_VER
+#include <windows.h>
+#else
+#include <pthread.h>
+#endif
+
+#include "flamegpu/model/ModelData.h"
+#include "flamegpu/simulation/CUDASimulation.h"
+#include "flamegpu/simulation/RunPlanVector.h"
+
+namespace flamegpu {
+namespace detail {
+
+AbstractSimRunner::AbstractSimRunner(const std::shared_ptr<const ModelData> _model,
+    std::atomic<unsigned int> &_err_ct,
+    std::atomic<unsigned int> &_next_run,
+    const RunPlanVector &_plans,
+    std::shared_ptr<const StepLoggingConfig> _step_log_config,
+    std::shared_ptr<const LoggingConfig> _exit_log_config,
+    int _device_id,
+    unsigned int _runner_id,
+    flamegpu::Verbosity _verbosity,
+    std::map<unsigned int, RunLog> &_run_logs,
+    std::queue<unsigned int> &_log_export_queue,
+    std::mutex &_log_export_queue_mutex,
+    std::condition_variable &_log_export_queue_cdn,
+    std::vector<ErrorDetail> &_err_detail,
+    const unsigned int _total_runners,
+    bool _isSWIG)
+      : model(_model->clone())
+      , device_id(_device_id)
+      , runner_id(_runner_id)
+      , total_runners(_total_runners)
+      , verbosity(_verbosity)
+      , err_ct(_err_ct)
+      , next_run(_next_run)
+      , plans(_plans)
+      , step_log_config(std::move(_step_log_config))
+      , exit_log_config(std::move(_exit_log_config))
+      , run_logs(_run_logs)
+      , log_export_queue(_log_export_queue)
+      , log_export_queue_mutex(_log_export_queue_mutex)
+      , log_export_queue_cdn(_log_export_queue_cdn)
+      , err_detail(_err_detail)
+      , isSWIG(_isSWIG) {
+}
+void AbstractSimRunner::start() {
+    this->thread = std::thread(&AbstractSimRunner::main, this);
+    // Attempt to name the thread
+#ifdef _MSC_VER
+    std::wstringstream thread_name;
+    thread_name << L"CUDASim D" << device_id << L"T" << runner_id;
+    // HRESULT hr =
+    SetThreadDescription(this->thread.native_handle(), thread_name.str().c_str());
+    // if (FAILED(hr)) {
+    //     fprintf(stderr, "Failed to name thread 'CUDASim D%dT%u'\n", device_id, runner_id);
+    // }
+#else
+    std::stringstream thread_name;
+    thread_name << "CUDASim D" << device_id << "T" << runner_id;
+    // int hr =
+    pthread_setname_np(this->thread.native_handle(), thread_name.str().c_str());
+    // if (hr) {
+    //     fprintf(stderr, "Failed to name thread 'CUDASim D%dT%u'\n", device_id, runner_id);
+    // }
```

Comment:
Lots of commented out partial code that should be cleaned up here.

Reply:
Again, this is code taken from `CUDAEnsemble`. It's useful for debugging if required (in future) hence why it was originally left it. I don't think i've ever seen it fail, but we don't want to spam users regardless.

Reply:
Fair enough, just a bit of a code smell. Grand scheme its fine either way.